src/library/serato/seratofeature.cpp

```diff
@@ -0,0 +1,1043 @@
+    Genre = 0x7467656e,          // tgen
+    Comment = 0x74636f6d,        // tcom
+    Grouping = 0x74677270,       // tgrp
+    Label = 0x746c626c,          // tlbl
+    Year = 0x74747972,           // ttyr
+    Length = 0x746c656e,         // tlen
+    Bitrate = 0x74626974,        // tbit
+    SampleRate = 0x74736d70,     // tsmp
+    Bpm = 0x7462706d,            // tbpm
+    DateAddedText = 0x74616464,  // tadd
+    DateAdded = 0x75616464,      // uadd
+    Key = 0x746b6579,            // tkey
+    BeatgridLocked = 0x6262676c, // bbgl
+    FileTime = 0x75746d65,       // utme
+    Missing = 0x626d6973,        // bmis
+    Sorting = 0x7472736f,        // osrt
+    ReverseOrder = 0x62726576,   // brev
+    ColumnTitle = 0x6f766374,    // ovct
+    ColumnName = 0x7476636e,     // tvcn
+    ColumnWidth = 0x74766377,    // tvcw
+    TrackPath = 0x7074726b,      // ptrk
+};
+
+struct serato_track_t {
+    QString filetype;
+    QString location;
+    QString title;
+    QString artist;
+    QString album;
+    QString genre;
+    QString comment;
+    QString grouping;
+    QString label;
+    int year = -1;
+    int duration = -1;
+    QString bitrate;
+    QString samplerate;
+    double bpm = -1.0;
+    QString key;
+    bool beatgridlocked = false;
+    bool missing = false;
+    quint32 filetime = 0;
+    quint32 datetimeadded = 0;
+};
+
+const QString kDatabaseDirectory = "_Serato_";
+const QString kDatabaseFilename = "database V2";
+const QString kCrateDirectory = "Subcrates";
+const QString kCrateFilter = "*.crate";
+const QString kSmartCrateDirectory = "Smart Crates";
+const QString kSmartCrateFilter = "*.scrate";
+
+const QString kSeratoLibraryTable = "serato_library";
+const QString kSeratoPlaylistsTable = "serato_library";
+const QString kSeratoPlaylistTracksTable = "serato_library";
+
+int createPlaylist(QSqlDatabase& database, QString name, QString databasePath) {
+    QSqlQuery query(database);
+    query.prepare(
+            "INSERT INTO serato_playlists (name, serato_db)"
+            "VALUES (:name, :serato_db)");
+    query.bindValue(":name", name);
+    query.bindValue(":serato_db", databasePath);
+
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query) << "databasePath: " << databasePath;
+        return -1;
+    }
+
+    return query.lastInsertId().toInt();
+}
+
+int insertTrackIntoPlaylist(QSqlDatabase& database, int playlistId, int trackId, int position) {
+    QSqlQuery query(database);
+    query.prepare(
+            "INSERT INTO serato_playlist_tracks (playlist_id, track_id, position) "
+            "VALUES (:playlist_id, :track_id, :position)");
+    query.bindValue(":playlist_id", playlistId);
+    query.bindValue(":track_id", trackId);
+    query.bindValue(":position", position);
+
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query);
+        return -1;
+    }
+
+    return query.lastInsertId().toInt();
+}
+
+inline QString parseText(const QByteArray& data, const quint32 size) {
```

Comment:
The function name is missleading because it is not actually a text parser. It coverts a "UTF-16BE" array to a QString. 
How about utf16BeToQString()

Reply:
It's named `parseText` because the  Serato DB string fields are prefixed with `t`, e. g. `tart` for the artist field. I figured that this probably stands for "text".

Reply:
I opted for `bytesToQString`, which is a bit easier on the eyes.